ui/media/js/main.js

```diff
@@ -354,15 +356,27 @@ function onUseAsInputClick(req, img) {
     maskSetting.checked = false
 }
 
-function onDownloadImageClick(req, img) {
-    const imgData = img.src
+function getDownloadFilename(img, suffix) {
     const imageSeed = img.getAttribute('data-seed')
     const imagePrompt = img.getAttribute('data-prompt')
     const imageInferenceSteps = img.getAttribute('data-steps')
     const imageGuidanceScale = img.getAttribute('data-guidance')
+    
+    return createFileName(imagePrompt, imageSeed, imageInferenceSteps, imageGuidanceScale, suffix)
+}
+
+function onDownloadJSONClick(req, img) {
+    const jsonDownload = document.createElement('a')
+    jsonDownload.download = getDownloadFilename(img, 'json')
+    jsonDownload.setAttribute('href', 'data:text/plain;charset=utf-8,' + encodeURIComponent(JSON.stringify(req, null, " ")))
```

Comment:
Can you please check if this should be `req.reqBody`? I'm not sure, but would be good to confirm that we're encoding the correct JSON object.

Reply:
The `req` object in the context of the button handler is the reqBody object, cf. https://github.com/cmdr2/stable-diffusion-ui/blob/beta/ui/media/js/main.js#L305:
```py
const req = Object.assign({}, reqBody, {
                seed: result?.seed || reqBody.seed
            })
```